lib/FieldLocation/FieldLocationFinal.js

```diff
@@ -1,115 +1,46 @@
-import React, {
-  useState,
-  useMemo,
-  useCallback,
-  useEffect,
-} from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
-import { Field } from 'react-final-form';
 
-import { stripesConnect } from '@folio/stripes/core';
 import { LocationLookup } from '@folio/stripes/smart-components';
-import {
-  IconButton,
-  TextField,
-} from '@folio/stripes/components';
 
-import { locationByPropManifest } from '../manifests';
+import { FieldSelectFinal } from '../FieldSelect';
 import { validateRequired } from '../utils';
+import { fieldSelectOptionsShape } from '../shapes';
 
 const FieldLocationFinal = ({
-  locationId,
   labelId,
-  mutator,
+  locationLookupLabel,
+  locationsForSelect,
   name,
-  onChange,
   required,
-}) => {
-  const [selectedLocation, setSelectedLocation] = useState({});
-
-  useEffect(() => {
-    if (locationId && selectedLocation.id !== locationId) {
-      mutator.fieldLocation.GET()
-        .then(setSelectedLocation);
-    }
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [locationId]);
-
-  const selectLocation = useCallback(
-    (location) => {
-      setSelectedLocation(location);
-
-      onChange(location);
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [onChange],
-  );
-
-  const clearLocation = useCallback(
-    () => {
-      setSelectedLocation({});
-
-      onChange({});
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [onChange],
-  );
-
-  const clearButton = useMemo(
-    () => {
-      if (selectedLocation.id) {
-        return (
-          <IconButton
-            icon="times-circle-solid"
-            onClick={clearLocation}
-            size="small"
-          />
-        );
-      }
-
-      return null;
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [selectedLocation],
-  );
-
-  return (
-    <div>
-      <Field
-        component={TextField}
-        disabled
-        endControl={clearButton}
-        format={() => selectedLocation.name}
-        fullWidth
-        hasClearIcon={false}
-        id={name}
-        label={labelId ? <FormattedMessage id={labelId} /> : ''}
-        marginBottom0
-        name={name}
-        required={required}
-        validate={required ? validateRequired : undefined}
-      />
-      <LocationLookup onLocationSelected={selectLocation} />
-    </div>
-  );
-};
+  selectLocationFromPlugin,
+}) => (
+  <div>
+    <FieldSelectFinal
+      dataOptions={locationsForSelect}
+      fullWidth
+      id={`field-${name}`}
+      label={labelId ? <FormattedMessage id={labelId} /> : ''}
```

Comment:
have you checked console for warnings? `FieldSelect` is different from the `Field`, there might be some warnings

Reply:
Checked it. No new error messages were found